app/src/main/java/nl/tudelft/bbw/API.java

```diff
@@ -32,75 +40,105 @@ private API() {
+        return owner;
     }
 
     /**
-     * Method to add a contact to your chain
+     * Getter method to get the API user's contact list
      *
-     * @param contact the contact
-     * @param signature byte array containing the signature
-     * @param message byte array containing the message
-     * @throws HashException               When creating a block results in an error
-     * @throws BlockAlreadyExistsException when adding a block results in an error
+     * @return List of blocks forming this user contact list.
      */
-    public static void addContactToChain(User contact, byte[] signature, byte[] message)
-            throws HashException, BlockAlreadyExistsException {
-        blockController.addBlockToChain(contact, signature, message);
+    public static List<Block> getMyContacts() {
+        return blockController.getBlocks(blockController.getOwnUser());
     }
 
     /**
-     * Method to revoke a contact to your chain
+     * Getter method to get the API user's name
      *
-     * @param contact the contact
-     * @throws HashException               When creating a block results in an error
-     * @throws BlockAlreadyExistsException when adding a block results in an error
+     * @return name of the API user
      */
-    public static void revokeContactFromChain(User contact)
-            throws HashException, BlockAlreadyExistsException {
-        blockController.revokeBlockFromChain(contact);
+    public static String getMyName() {
+        return blockController.getOwnUser().getName();
     }
 
     /**
-     * Method to add a contact to your database
+     * Getter method to get the API user's Iban
+     *
+     * @return Iban number of the API user
+     */
+    public static String getMyIban() {
+        return blockController.getOwnUser().getIban();
+    }
+
+    /**
+     * Getter method to get the API user's Public Key
+     *
+     * @return Public Key  of the API user
+     */
+    public static EdDSAPublicKey getMyPublicKey() {
+        return blockController.getOwnUser().getPublicKey();
+    }
+
+
+    /**
+     * Method to add a contact to your chain
      *
-     * @param owner   the User
-     * @param contact the contact
      * @throws HashException               When creating a block results in an error
      * @throws BlockAlreadyExistsException when adding a block results in an error
      */
-    public static void addContactToDatabase(User owner, User contact)
-            throws HashException, BlockAlreadyExistsException {
-        blockController.createKeyBlock(owner, contact);
+    public static void addAcquaintance(Acquaintance acquaintance)
+            throws HashException, BlockAlreadyExistsException, NoSuchAlgorithmException, InvalidKeyException, SignatureException {
+
+        final byte[] message = acquaintance.getPublicKey().getEncoded();
+        final byte[] signature = ED25519.generateSignature(message, blockController.getOwnUser().getPrivateKey());
```

Comment:
You have to generate the signature in the activity. In the API, the signature only gets verified.